### PR TITLE
fix(import-flow): updates gitHub screen as per UXD

### PR DESCRIPTION
**Pull request by invincibleJai**
This PR fixes : 

- Updates GitHub screen as per UXD for Import/Quickstarts
- Introduces typeahead for repository select box in Import flow
- Removes Optional from pipelines
- Fixes broken test in project-summary
- Change repo status check flow (for existing repo) in Quickstart
- Adds tests for component of Import flow

### CODE REVIEW

**Comment by invincibleJai** [comment]
@dlabrecq @arunkumars08 @jyasveer Pls review it

File: src/app/launcher/create-app/gitprovider-createapp-step/gitprovider-createapp-step.component.less [hunk 1]
```diff
@@ -10,12 +10,15 @@
     &-icon {
       min-height: 300px;
       border-right: 1px solid;
+      margin: 10px 0px;
```
Comment (arunkumars08) [inline src/app/launcher/create-app/gitprovider-createapp-step/gitprovider-createapp-step.component.less]: just 0 should be fine.

Comment (invincibleJai) [reply src/app/launcher/create-app/gitprovider-createapp-step/gitprovider-createapp-step.component.less]: sure

File: src/app/launcher/create-app/gitprovider-createapp-step/gitprovider-createapp-step.component.html [hunk 2]
```diff
@@ -24,33 +27,35 @@ <h2>GitHub</h2>
...
-            </span>
-            <button class="btn btn-primary btn-lg f8launcher-authorize-account"
-                    [disabled]="launcherComponent.summary?.gitHubDetails?.authenticated === true"
-                    (click)="connectAccount($event)">Log In &amp; Authorize Account</button>
-          </div>
+          <h3>Authorized Account Information</h3>
           <div class="form-horizontal">
             <div class="form-group">
-              <label for="ghOrg" class="col-sm-2 control-label">Organization</label>
+              <label for="ghAccount" class='col-sm-2 control-label'>Username</label>
```
Comment (mindreeper2420) [inline src/app/launcher/create-app/gitprovider-createapp-step/gitprovider-createapp-step.component.html]: please update the single quotes to double quotes around the class names

File: src/app/launcher/create-app/gitprovider-createapp-step/gitprovider-createapp-step.component.less [hunk 3]
```diff
@@ -27,12 +30,11 @@
       padding-left: 15px;
       padding-right: 15px;
       min-height: 300px;
+      margin: 10px 0px;
```
Comment (mindreeper2420) [inline src/app/launcher/create-app/gitprovider-createapp-step/gitprovider-createapp-step.component.less]: same as above - `margin: 10px 0;`

File: src/app/launcher/import-app/gitprovider-importapp-step/gitprovider-importapp-step.component.less [hunk 4]
```diff
@@ -7,35 +7,41 @@
...
-        display: inline-flex;
-        justify-content: flex-end;
-      }
-    }
     &-icon {
       min-height: 300px;
-      background-color: @color-pf-black-200;
-      text-align: center;
+      border-right: 1px solid;
+      margin: 10px 0px;
```
Comment (mindreeper2420) [inline src/app/launcher/import-app/gitprovider-importapp-step/gitprovider-importapp-step.component.less]: please update to `margin: 10px 0;`

File: src/app/launcher/import-app/gitprovider-importapp-step/gitprovider-importapp-step.component.less [hunk 5]
```diff
@@ -7,35 +7,41 @@
...
+      text-align: left;
+    }
     &-information {
-      padding-top: 25px;
-      padding-right: 15px;
-      padding-bottom: 25px;
       padding-left: 15px;
+      padding-right: 15px;
       min-height: 300px;
+      margin: 10px 0px;
```
Comment (mindreeper2420) [inline src/app/launcher/import-app/gitprovider-importapp-step/gitprovider-importapp-step.component.less]: `margin: 10px 0;`

**Comment by invincibleJai** [comment]
Thanks @mindreeper2420 @arunkumars08, pls take a look

**Pull request merged**